src/components/spreadsheet/table-wrapper.js

```diff
@@ -238,6 +240,75 @@ const TableWrapper = (props) => {
     const { updateFilter, filterSelector, initFilters } =
         useAggridLocalRowFilter(gridRef, equipementFiltersSelectorKeys);
 
+    const equipmentDefinition = useMemo(
+        () => ({
+            type: TABLES_DEFINITION_INDEXES.get(tabIndex).type,
+            fetchers: TABLES_DEFINITION_INDEXES.get(tabIndex).fetchers,
+        }),
+        [tabIndex]
+    );
+
+    const formatFetchedEquipmentHandler = useCallback(
+        (fetchedEquipment) => {
+            return formatFetchedEquipment(
+                equipmentDefinition.type,
+                fetchedEquipment
+            );
+        },
+        [equipmentDefinition.type]
+    );
+
+    const formatFetchedEquipmentsHandler = useCallback(
+        (fetchedEquipments) => {
+            //Format the equipments data to set calculated fields, so that the edition validation is consistent with the displayed data
+            return formatFetchedEquipments(
+                equipmentDefinition.type,
+                fetchedEquipments
+            );
+        },
+        [equipmentDefinition.type]
+    );
+
+    const { equipments, errorMessage, isFetching } = useSpreadsheetEquipments(
+        equipmentDefinition,
+        formatFetchedEquipmentsHandler
+    );
+
+    // Function to get the columns that have isEnum filter set to true in customFilterParams
+    const getEnumFilterColumns = useCallback(() => {
+        const generatedTableColumns =
+            TABLES_DEFINITION_INDEXES.get(tabIndex).columns;
+        return generatedTableColumns.filter(
+            (column) =>
+                column.customFilterParams &&
+                column.customFilterParams.isEnum === true
+        );
+    }, [tabIndex]);
+
+    const generateEquipmentsFilterEnums = useCallback(() => {
+        if (!equipments) {
+            return {};
+        }
+        const enumFilterColumns = getEnumFilterColumns();
+        const filterEnums = {};
+        enumFilterColumns.forEach((column) => {
+            const columnValues = new Set();
+            equipments.forEach((equipment) => {
+                const columnValue = deepFindValue(equipment, column.field);
+                if (columnValue !== null && columnValue !== undefined) {
+                    columnValues.add(columnValue);
+                }
+            });
+            filterEnums[column.field] = Array.from(columnValues);
+        });
+        return filterEnums;
+    }, [getEnumFilterColumns, equipments]);
+
+    const filterEnums = useMemo(
+        () => generateEquipmentsFilterEnums(),
+        [generateEquipmentsFilterEnums]
+    );
```

Comment:
call method directly in "enrichColumn" rather than memoizing it here ?

Reply:
The 'enrich column' function is invoked for each defined column. If I proceed in that manner, I will end up calculating the enum list as many times as there are columns in the table

Reply:
I agree it's better this way then